src/main/java/ch/wisv/chue/HueService.java

```diff
@@ -88,14 +99,23 @@ public void loadEvent(HueEvent event, int duration, String... lightIdentifiers)
      * @param lightIdentifiers the lights to strobe
      */
     public void strobe(int millis, String... lightIdentifiers) {
-        hueFacade.strobe(millis, getLightIdentifiers(lightIdentifiers));
+        if (!hueFacade.bridgeAvailable()) {
+            throw new EventNotExecutedException("Hue bridge is not available");
+        }
+
+        try {
+            hueFacade.strobe(millis, getLightIdentifiers(lightIdentifiers));
+        } catch (BridgeUnavailableException e) {
+            e.printStackTrace();
```

Comment:
You're right, but the problem is that you want to make `strobe` defensive (i.e. throw the exception if there's no bridge), and at the same time you want to check before you execute. We could collapse this into

```java
try {
  hueFacade.strobe(millis, getLightIdentifiers(lightIdentifiers));
} catch (BridgeUnavailableException e) {
  throw new EventNotExecutedException(e.getMessage());
}
```

What do you think? (Apart from the exception embedding instead of copying the message)